code/datums/components/cell.dm

```diff
@@ -16,49 +16,65 @@
 	/// If draining charge on process(), how much to drain per process call
 	var/charge_drain = 10
 	/// If the parent should show cell charge on examine
-	var/display_charge = TRUE
+	var/display_charge = FALSE
 	/// From how many tiles at the highest someone can examine the parent to see the charge
 	var/charge_examine_range = 1
 	/// If the component requires a cell to be inserted to work instead of having an integrated one
 	var/cell_insert = FALSE
 	/// Ref to an inserted cell. Should only be null if cell_insert is false
 	var/obj/item/cell/inserted_cell
+	/// What should be displayed as the examine string if display_charge is TRUE. %CHARGE% and %MAXCHARGE% will be replaced with the remaining charge in the cell and its maximum, respectively
+	var/examine_string = "A small gauge in the corner reads \"Power: %CHARGE%\"."
 
 
 /datum/component/cell/Initialize(
 	max_charge = 10000,
 	hit_charge = FALSE,
 	max_recharge_tick = 400,
 	charge_drain = 10,
-	display_charge = TRUE,
+	display_charge = FALSE,
 	charge_examine_range = 1,
 	cell_insert = FALSE,
+	cell_insert_default_cell = /obj/item/cell,
+	examine_string = "A small gauge in the corner reads \"Power: %CHARGE%\".",
+	initial_charge = -1,
 	)
 
 	. = ..()
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
 
 	src.max_charge = max_charge
-	charge = max_charge
+	if(initial_charge == -1)
+		charge = max_charge
+	else
+		charge = min(initial_charge, max_charge)
 	src.hit_charge = hit_charge
 	src.max_recharge_tick = max_recharge_tick
 	src.charge_drain = charge_drain
 	src.display_charge = display_charge
 	src.charge_examine_range = charge_examine_range
 	src.cell_insert = cell_insert
+	if(cell_insert)
+		inserted_cell = new cell_insert_default_cell(parent)
+	src.examine_string = examine_string
```

Comment:
you could use `src.examine_string = examine_string || src.examine_string` (or just a if) so you don't have to duplicate the default string in arguments above